lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,230 @@
+const { Compilation, sources } = require('webpack');
+const parse5 = require('parse5');
+const {
+  getParagonVersion,
+  getParagonThemeCss,
+  replacePeriodsWithHyphens,
+} = require('../../../config/data/paragonUtils');
+
+const paragonVersion = getParagonVersion(process.cwd());
+const paragonThemeCss = getParagonThemeCss(process.cwd());
+
+const brandVersion = getParagonVersion(process.cwd(), { isBrandOverride: true });
+const brandThemeCss = getParagonThemeCss(process.cwd(), { isBrandOverride: true });
+
+class ParagonWebpackPlugin {
+  constructor() {
+    this.paragon = {
+      version: paragonVersion,
+      coreEntryName: undefined,
+      themeVariantEntryNames: {},
+    };
+
+    this.brand = {
+      version: brandVersion,
+      coreEntryName: undefined,
+      themeVariantEntryNames: {},
+    };
+
+    if (!paragonThemeCss && !brandThemeCss) {
+      return;
+    }
+
+    if (paragonThemeCss) {
+      // Core Paragon
+      this.paragon.coreEntryName = replacePeriodsWithHyphens(paragonThemeCss.core.entryName);
```

Comment:
Doesn't this return the same hyphened string that already exists inside `paragonThemeCss.core` under `outputChunkName` property?
I think all usages of `replacePeriodsWithHyphens` can be removed from this file in favour of using `outputChunkName` property returned by `getParagonThemeCss` function unless I'm missing something.

Also, if you do that I feel like you can either significantly simplify `construct` method or just remove it altogether because `paragonThemeCss` and `brandThemeCss` already contain all of the data you put into `paragon` and `brand` objects, so you can just use them in the class instead.